src/storage/build.ts

```diff
@@ -84,27 +138,17 @@ export function buildStorage({ set, find, remove }: BuildStorage): AxiosStorage
         return { state: 'empty' };
       }
 
-      if (
-        // Not cached or fresh value
-        value.state !== 'cached' ||
-        !isExpired(value)
-      ) {
-        return value;
-      }
-
-      if (canStale(value)) {
-        const stale: StaleStorageValue = {
-          state: 'stale',
-          createdAt: value.createdAt,
-          data: value.data
-        };
-
-        await set(key, stale, config);
-        return stale;
+      switch (value.state) {
```

Comment:
This switch don't fits well in this code, You can reuse some of repeated statements on `handleCachedValues` and `handleStaleValues`.

Try something like:


```ts
if(value.state === 'empty' || value.state !== 'loading') {
 return value;
}

// Non expired stale or cached are fine
if (!isExpired(value)) {
 return value;
}

// Updates to stale if possible
if(value.state === 'cached' && canStale(value)) {
 value={
  state: 'stale',
  createdAt: value.createdAt,
  data: value.data,
  ttl: value.staleTtl !== undefined ? value.staleTtl + value.ttl : undefined };

  await set(key, value, config);


// Maybe the newly value are already expired
  if (!isExpired(value)) {
      return value;
    }

}


// Can overstale because of some meta values
    if (hasUniqueIdentifierHeader(value)) {
      return value;
  }


// Cache is not valid anymore.
await remove(key, config);
return { state: 'empty' };
```

I did write this in the github comments and maybe i'm crazy bcs of that, but I think you got the idea that we can change these 2 functions declaration and a switch to just 2 if statements.